tests/neg-custom-args/fatal-warnings/i4674.scala

```diff
@@ -0,0 +1,9 @@
+object i4674 {
+
+  val x: String = "test"
+
+  x.foreach {
+    case 's' => println("s")
+    case c: Char => println(c)
```

Comment:
You will need to add `// error: type test always succeed` at the end of the line to make the test succeed